applications/luci-app-vnstat2/htdocs/luci-static/resources/view/vnstat2/graphs.js

```diff
@@ -1,9 +1,58 @@
 // This is free software, licensed under the Apache License, Version 2.0
 
 'use strict';
+'require poll';
 'require view';
 'require fs';
 'require ui';
+'require uci';
+'require rpc';
+
+var RefreshIfaces = "";
+var RefreshTabs = ['s', 't', '5', 'h', 'd', 'm', 'y'];
+
+var callServiceList = rpc.declare({
+	object: 'service',
+	method: 'list',
+	params: [ 'name' ],
+	expect: { '': {} }
+});
+
+function RefreshGraphs() {
+	RefreshTabs.forEach(function (id) {
+		RefreshIfaces.forEach(function (iface) {
+			fs.exec_direct('/usr/bin/vnstati', [ '-' + id, '-i', iface, '-o', '-' ], 'blob').then(function(res) {
+				document.getElementById('graph_' + id + '_' + iface).src = URL.createObjectURL(res);
+			});
+		});
+	});
+}
+
+function IfacesResetData(ev) {
+	ui.showModal(_('Delete data for ALL interfaces'), [
+		E('p', _('The data will be removed from the database permanently. This cannot be undone.')),
+		E('div', { 'class': 'right' }, [
+			E('div', {
+				'class': 'btn',
+				'click': L.hideModal
+			}, _('Cancel')),
+			' ',
+			E('div', {
+				'class': 'btn cbi-button-negative',
+				'click': function(ev) {
+					RefreshIfaces.forEach(function (iface) {
+						fs.exec_direct('/usr/bin/vnstat', [ '--remove', '-i', iface, '--force' ], 'blob').then(function() {
+							fs.exec_direct('/usr/bin/vnstat', [ '--add', '-i', iface ], 'blob').then(function() {
```

Comment:
This will silently fail if the user only has read-only permission. The ACL only allows calls to `vnstat --remove` and `vnstat --add` if the user has full access. It probably makes sense to disable this button when there is only read-only access (similar to the delete button in the config page).